examples/getting_started/quickstart.ipynb

```diff
@@ -0,0 +1,626 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Translate the following to {language}: {text}\")\n",
+    "def translate(text: str, language: str):\n",
+    "    return {\"temperature\": 0.7 if language == \"French\" else 0.5}\n",
+    "\n",
+    "\n",
+    "response = translate(\"Hello, world!\", \"French\")\n",
+    "print(response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we're dynamically setting the `temperature` parameter based on the target language.\n",
+    "\n",
+    "Now, let's look at an example of chaining:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Summarize the following text: {text}\")\n",
+    "def summarize(text: str):\n",
+    "    pass\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Translate the following to French: {text}\")\n",
+    "def translate_to_french(text: str):\n",
+    "    pass\n",
+    "\n",
+    "\n",
+    "original_text = \"Recently, it's been really hot, so I’m thinking of gathering with friends to enjoy some champagne outdoors. I’m looking for something refreshing and light, especially a champagne that’s enjoyable even on a hot day. If possible, I’d love to know about any recommended champagne that’s perfect for relaxing with everyone.\"\n",
+    "summary = summarize(original_text)\n",
+    "french_summary = translate_to_french(summary.content)\n",
+    "print(french_summary.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This chain first summarizes a text, then translates the summary to French, demonstrating how you can combine multiple LLM calls to perform more complex tasks."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## JSON Mode\n",
+    "\n",
+    "JSON mode allows you to directly parse LLM outputs as JSON. This is useful when you need structured data output.\n",
+    "\n",
+    "Here's an example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@openai.call(\"gpt-4o-mini\", json_mode=True)\n",
+    "@prompt_template(\"Provide information about {city} in JSON format\")\n",
+    "def city_info(city: str):\n",
+    "    pass\n",
+    "\n",
+    "\n",
+    "response = city_info(\"Tokyo\")\n",
+    "print(response.content)  # This will be a JSON-formatted string"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By setting `json_mode=True`, we instruct the LLM to format its response as JSON, which can then be easily parsed and used in your application."
```

Comment:
I think it's important to note here that not all providers have an explicit JSON mode, and for those providers we simply attempt to instruct the model to provide JSON; however, there is no guarantee that it will actually output only JSON (as it may start with some text like "Here is the JSON: ..."). This is where Output Parsers can be useful (and works as a great next section following such a note).

Reply:
Your explanation makes a lot of sense and makes sense to me, so I will quote it.